pkg/machines/components/vm/vmMigrateDialog.jsx

```diff
@@ -0,0 +1,194 @@
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+import React, { useState } from 'react';
+import {
+    Button,
+    Checkbox,
+    Form,
+    FormGroup,
+    Modal,
+    Radio,
+    TextInput,
+    Tooltip
+} from '@patternfly/react-core';
+
+import { migrateToUri } from '../../libvirt-dbus.js';
+import { isEmpty, isObjectEmpty } from '../../helpers.js';
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+
+const _ = cockpit.gettext;
+
+const DestUriRow = ({ validationFailed, destUri, setDestUri }) => {
+    return (
+        <FormGroup label={_("Destination URI")} fieldId="dest-uri"
+                   id="dest-uri"
+                   validated={validationFailed.destUri ? "error" : "default"}>
+            <TextInput id='dest-uri-input'
+                       validated={validationFailed.destUri ? "error" : "default"}
+                       value={destUri}
+                       placeholder={cockpit.format(_("Example, $0"), "qemu+ssh://192.0.2.16/system")}
+                       onChange={setDestUri} />
+        </FormGroup>
+    );
+};
+
+const TypeRow = ({ type, setType, setUndefineSource, setStorage }) => {
+    return (
+        <FormGroup label={_("Migrated VM state")}
+                   fieldId="type"
+                   id="type"
+                   hasNoPaddingTop>
+            <Radio id="live"
+                   name="type"
+                   label={_("Live")}
+                   isChecked={type === "live"}
+                   onChange={() => setType("live")} />
+            <Radio id="offline"
+                   name="type"
+                   label={_("Shut off")}
+                   isChecked={type === "offline"}
+                   onChange={() => {
+                       setType("offline");
+                       setStorage("shared");
+                   }} />
+        </FormGroup>
+    );
+};
+
+const UndefineSourceRow = ({ undefineSource, setUndefineSource }) => {
+    return (
+        <FormGroup label={_("Delete source VM")}
+                   fieldId="undefine-source"
+                   id="undefine-source"
+                   hasNoPaddingTop>
+            <Checkbox id="undefine-source"
+                      isChecked={undefineSource}
+                      label={_("Remove virtual machine from this host")}
+                      onChange={setUndefineSource} />
+        </FormGroup>
+    );
+};
+
+const StorageRow = ({ storage, setStorage, type }) => {
+    const copyDisabled = type === "offline";
+    let copyRadio = (
+        <Radio id="copy"
+               name="source"
+               label={_("Copy storage")}
```

Comment:
I think we need to add a explanatory popover here: "I checked all non-shared non-readonly images will be transferred."